spec/reek/smells/too_many_constants_spec.rb

```diff
@@ -0,0 +1,115 @@
+require_relative '../../spec_helper'
+require_lib 'reek/smells/too_many_constants'
+require_relative 'smell_detector_shared'
+
+RSpec.describe Reek::Smells::TooManyConstants do
+  context 'counting constants' do
+    it 'should not report for non-excessive constants' do
+      src = <<-EOS
+        class Dummy
+          A = B = C = D = E = 1
+        end
+      EOS
+
+      expect(src).not_to reek_of(:TooManyConstants)
+    end
+
+    it 'should count each constant only once' do
+      src = <<-EOS
+        class Good
+          A = B = C = D = E = 1
+        end
+
+        class Bad
+          A = B = C = D = E = 1
+        end
+
+        class Ugly
+          A = B = C = D = E = 1
+        end
+      EOS
+
+      expect(src).not_to reek_of(:TooManyConstants)
+    end
+
+    it 'should report for excessive constants inside classes' do
+      src = <<-EOS
+        class Dummy
+          A = B = C = D = E = 1
+          F = 1
+        end
+      EOS
+
+      expect(src).to reek_of(:TooManyConstants)
+    end
+
+    it 'should not account class definition' do
+      src = <<-EOS
+        module Dummiest
+          class Dummy
+            A = B = C = D = E = 1
+
+            Error = Class.new(StandardError)
+          end
+        end
+      EOS
+
+      expect(src).to_not reek_of(:TooManyConstants)
+    end
+
+    it 'should not account struct definition' do
+      src = <<-EOS
+        module Dummiest
+          class Dummy
+            A = B = C = D = E = 1
+
+            Struct = Struct.new
+          end
+        end
+      EOS
+
+      expect(src).to_not reek_of(:TooManyConstants)
+    end
+
+    it 'should report for excessive constants inside modules' do
+      src = <<-EOS
+        module Dummiest
+          A = B = C = D = E = 1
+          F = 1
+
+          class Dummy
+          end
+        end
+      EOS
+
+      expect(src).to reek_of(:TooManyConstants, context: 'Dummiest')
+    end
+  end
+
+  context 'smell report' do
+    src = <<-EOS
+      module Moduly
+        class Klass
+          A = B = C = D = E = 1
+          F = 1
+        end
+      end
+    EOS
+
+    it 'reports the number of consts' do
+      expect(src).to reek_of(:TooManyConstants, count: 6)
+    end
+
+    it 'reports the line where it occurs' do
+      expect(src).to reek_of(:TooManyConstants, lines: [1])
+    end
+
+    it 'reports the line where it occurs' do
+      expect(src).to reek_of(:TooManyConstants, context: 'Moduly::Klass')
```

Comment:
As the class is also inside module both the class and the module is going to smell.